test/verifier/embedded-zkp-verifier.test.ts

```diff
@@ -2,11 +2,18 @@ import { expect } from "chai";
 import { DeployHelper } from "../../helpers/DeployHelper";
 import { ethers } from "hardhat";
 import { packValidatorParams } from "../utils/validator-pack-utils";
-import { prepareInputs, publishState } from "../utils/state-utils";
+import { prepareInputs } from "../utils/state-utils";
 import { Block, Signer } from "ethers";
+import {
+  packCrossChainProofs,
+  packGlobalStateUpdateWithSignature,
+  packIdentityStateUpdateWithSignature,
+  packZKProof,
+} from "../utils/packData";
+import proofJson from "../validators/sig/data/valid_sig_user_genesis.json";
```

Comment:
As long as you have Groth16VerifierValidatorStub now you can put any public inputs without invalidation groth16 proof. 
In That way you can avoid messages like "id type is not supported",  "Challenge should match the sender", ...